auth/src/main/kotlin/gov/cdc/prime/reportstream/auth/service/ProxyURIStrategy.kt

```diff
@@ -0,0 +1,56 @@
+package gov.cdc.prime.reportstream.auth.service
+
+import gov.cdc.prime.reportstream.auth.config.ApplicationConfig
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Profile
+import org.springframework.stereotype.Component
+import java.net.URI
+
+/**
+ * Implementations are ways to decide the ultimate destination of an incoming request
+ */
+interface ProxyURIStrategy {
+    fun getTargetURI(incomingUri: URI): URI
+}
+
+/**
+ * This implementation decides via the path prefix. Currently used locally for when all services are
+ * running on different ports of localhost.
+ *
+ * Configured under proxyConfig.pathMappings
+ *
+ * http://localhost:9000/submissions/health -> http://localhost:8880/health
+ */
+@Component
+@Profile("local")
+class PathPrefixProxyURIStrategy(
+    private val applicationConfig: ApplicationConfig,
+) : ProxyURIStrategy {
+    override fun getTargetURI(incomingUri: URI): URI {
+        val proxyPathMappings = applicationConfig.proxyConfig.pathMappings
+        val maybePathMapping = proxyPathMappings.find { incomingUri.path.startsWith(it.pathPrefix) }
+        return if (maybePathMapping != null) {
+            val baseUri = URI(maybePathMapping.baseUrl)
+            val path = incomingUri.path.removePrefix(maybePathMapping.pathPrefix)
+            URI(
+                baseUri.scheme,
+                baseUri.userInfo,
+                baseUri.host,
+                baseUri.port,
+                path,
+                incomingUri.query,
+                incomingUri.fragment
+            )
+        } else {
+            throw IllegalStateException("no configured proxy target in path mappings for path=${incomingUri.path}")
```

Comment:
how is this - and for that matter any - unexpected exception handled? If memory serves spring-boot will return an html document to the user. If that's what's going on and that's not what we want - you can use @RestControllerAdvice to create a global catch-all to ensure all exceptions result in a consistent return value protocol to the user.

Reply:
Right now it returns a default Spring json response. It's definitely in my todo list for the app to make that better but figured it was ok for now.